### DISCUSSION TITLE
Use subtemplate

### DISCUSSION THREAD

**Discussion by Tindome** [hairyhenderson/gomplate#1014] [Q&A] [answered]
Hello,

I want to add a subtemplate file to emulate boilr or pythonic equivalent (cookiecutter)

thus i create a template like this

____________________________________ mainfile.tpl
...
#-------------------
#Includes
#-------------------

{{ define "color_definition" -}}
{{- file.Read "./tpl/color_definition.tpl" }}
{{- end -}}

{{ if (datasource "sessions").USECOLORLIB -}}
#--------------------
#Color definitions
#--------------------
WD := $(shell pwd -P)
MK_INC := $(WD)/mklibs
include $(WD)/../../Makefile/mklibs/Colors #old is $(MK_INC)/Colors
{{- end }}

#same but with inclusion of external

{{ tmpl.Exec "color_definition" (datasource "sessions") }}
.....

____________________________________ color_definition.tpl
...
#--------------------
#Color definitions
#--------------------
WD := $(shell pwd -P)
MK_INC := $(WD)/mklibs
include $(WD)/../../Makefile/mklibs/Colors #old is $(MK_INC)/Colors

{{ .USECOLORLIB }}

______________________________________ param.json
{
	"USECOLORLIB": true,
}

````
$>gomplate_linux-amd64-slim -f ./makefile.tpl  -o makefile -d sessions=./params.json
````

As far as i could guess the problem is in the file.Read part, any though about doing this another way ?

Thanks

**Comment by hairyhenderson** [comment, marked as answer]
Hi @Tindome, thanks for logging this. I've converted it to a discussion since it fits better here.

I don't think you need to use `file.Read` or `tmpl.Exec` here at all - it looks like the [external nested templates](https://docs.gomplate.ca/syntax/#external-templates) feature (i.e. [`--template`/`-t`](https://docs.gomplate.ca/usage/#template-t)) should work for you:

```console
$ gomplate -t color_definition=./tpl/color_definition.tpl -d sessions=./params.json -f makefile.tpl -o makefile
```

Does that help?

**Reply by Tindome** [reply]
yes thanks a lot, i ve seen it before but not really understand the way to use it
